sunpy/io/_fits.py

```diff
@@ -99,6 +100,15 @@ def read(filepath, hdus=None, memmap=None, **kwargs):
                 message += repr(e)
                 warn_user(message)
 
+            if memmap == True and not (isinstance(pairs[0].data.base, mmap.mmap) and pairs[0].data.base is not None):
+                try:
+                    if pairs[0].header['BZERO'] != 0:
+                        warn_user("Data array is not memory mapped. Use 'disable_image_compression=True' to preserve memory mapping.")
```

Comment:
You can preserve the memmap, but it would be entirely useless as you wont be able to see the image any more? I don't think this is a good warning to throw.

Reply:
All that's true, but I don't think that this warning is helpful as written for someone who doesn't have a good understanding of FITS internals. I can just see someone doing that and then being like waaaat

Reply:
The easy "fix" would be to elaborate in the warning, e.g.:
> Data array is not memory mapped because it is stored in the file using compression. Specify `disable_image_compression=True` to preserve memory mapping, but only if you do not need the data array to be decompressed.